### ISSUE TITLE
Bad indentation in function applications with parenthesized args

### ISSUE THREAD

**Issue by osa1** [rust-lang/rust.vim#77]
Indented properly:

```rust
fn test() {
    f(x, y,
      z);
}
```

Indentation gets weird if I wrap `x, y` with parens:

```rust
fn test() {
    f((x, y),
    z);
}
```

**Comment by chris-morgan** [comment]
Yeah, known issue which is pretty much unavoidable while we’re using only vimscript for it and not a proper parser. I’d be interested in whether we could integrate rustfmt into it in some way. You’d want some sort of persistent background service to do it, like how YouCompleteMe works, or performance will be too bad.

**Comment by osa1** [comment]
Too bad this is hard to fix. It gets annoying real quick as function calls also break it:

```rust
fn test() {
  f(x(),
  y);
}
```

**Comment by taralx** [comment]
How does the C indenter do it?
